### PR TITLE
[Fit API] update estimator

### PR DESCRIPTION
**Pull request by roywei**
## Description ##
cc @szha
1. clean up context checking logic
2. remove trailing numbers in loss name to avoid confusion in logging

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/gluon/contrib/estimator/estimator.py [hunk 1]
```diff
@@ -88,12 +88,17 @@ def _check_metrics(self, metrics):
...
-            context = [context]
-        elif isinstance(context, list) and all([isinstance(c, Context) for c in context]):
-            context = context
-        elif not context:
+        if context:
+            # check context values, only accept Context or a list of Context
+            if isinstance(context, Context):
+                context = [context]
+            elif isinstance(context, list) and all([isinstance(c, Context) for c in context]):
+                context = context
```
Comment (szha) [inline python/mxnet/gluon/contrib/estimator/estimator.py]: should we check for the GPU device index too? also, try querying `num_gpus()` only once.

Comment (roywei) [reply python/mxnet/gluon/contrib/estimator/estimator.py]: @szha thanks! I m now asserting context must be in `available_context` which is `[cpu(), gpu(0), ...., gpu(num_gpus-1)]`.

Comment (szha) [reply python/mxnet/gluon/contrib/estimator/estimator.py]: cpu(65536) is actually valid in mxnet regardless of the number of physical CPUs

Comment (roywei) [reply python/mxnet/gluon/contrib/estimator/estimator.py]: How about:
```
available_context = [mx.gpu(i) for i in range(num_gpus)]
assert ctx  in available_context or str(ctx).startswith('cpu')
```

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Pull request merged**